cmake/dca_compiler_options.cmake

```diff
@@ -21,7 +21,7 @@
 set(DCA_WARNINGS -Wall -Wextra -Wpedantic -Wno-sign-compare -Wno-dangling-else)
 
 # Languange standard
-set(DCA_STD_FLAG -std=c++14)
+set(DCA_STD_FLAG -std=c++17)
```

Comment:
Forgot this one. In the end I only need to not make a separate .cpp file for a single static variable of  CudaProfiler. I can make it, but we might also consider the upgrade given that all the dependencies on Summit are met.